app/controllers/users_controller.rb

```diff
@@ -19,26 +19,14 @@ def new_with_invitation_token
 
   def create
     @user = User.new(user_params)
+    result = UserSignup.new(@user).sign_up(params[:stripeToken], params[:invitation_token])
 
-    if @user.valid?
-      charge = StripeWrapper::Charge.create(
-        amount: 999,
-        source: params[:stripeToken],
-        description: "Sign up charge for #{@user.email}"
-      )
-      if charge.successful?
-        @user.save
-        handle_invitation if params[:invitation_token].present?
-        AppMailer.send_welcome_email(@user).deliver
-        flash[:success] = "You have registered successfully"
-        session[:user_id] = @user.id
-        redirect_to home_path
-      else
-        flash.now[:error] = charge.error_message
-        render :new and return
-      end
+    if result.successful?
+      flash[:success] = "You have registered successfully"
+      session[:user_id] = @user.id
+      redirect_to home_path
     else
-      flash.now[:error] = "Please fix the following errors"
+      flash[:error] = result.error_message
```

Comment:
`flash.now` was correct solution. with render - its always `flash.now` I am surprised you have changed that.